examples/island/Island.cpp

```diff
@@ -123,14 +123,16 @@ int main()
     {
         statusText.setString("Shaders and/or Vertex Buffers Unsupported");
     }
-    else if (!terrainShader.loadFromFile("resources/terrain.vert", "resources/terrain.frag"))
+    else if (!(terrainShader = sf::Shader::loadFromFile("resources/terrain.vert", "resources/terrain.frag")))
     {
         prerequisitesSupported = false;
```

Comment:
Get rid of this `prerequisitesSupported` variable altogether, and inline the check on line 122. Then before using the shader you can check if `terrainShader.has_value`, so the shader itself basically tells you whether it's available or not.